src/shared/components/CSVExportButton.tsx

```diff
@@ -1,72 +1,102 @@
 // Libraries
 import React, {PureComponent} from 'react'
-import {connect} from 'react-redux'
+import {connect, ConnectedProps} from 'react-redux'
 
 // Components
-import {Button, ComponentStatus, IconFont} from '@influxdata/clockface'
-
-// Utils
-import {downloadTextFile} from 'src/shared/utils/download'
-import {getActiveTimeMachine} from 'src/timeMachine/selectors'
-import {createDateTimeFormatter} from 'src/utils/datetime/formatters'
+import {
+  Button,
+  ComponentStatus,
+  ComponentColor,
+  IconFont,
+} from '@influxdata/clockface'
+
+// Selectors and Actions
+import {getActiveQuery} from 'src/timeMachine/selectors'
+import {
+  runDownloadQuery,
+  DOWNLOAD_EVENT_COMPLETE,
+} from 'src/timeMachine/actions/queries'
 
 // Types
-import {AppState} from 'src/types'
+import {AppState, RemoteDataState} from 'src/types'
+
+type Props = ConnectedProps<typeof connector>
 
-interface StateProps {
-  files: string[] | null
+interface State {
+  state: RemoteDataState
```

Comment:
Yes, we count set the class prop directly for state.

I was electing to follow standardized react `setState()` approach because we (1) use this approach in this legacy part of the codebase, and (2) it always retriggers a re-render per react's listeners.

Reply:
In the two places where you call `setState`, you're doing it with `setState({state: ...})`. Why not just `setState(...)`, since it's a single enum value?

Reply:
React API wants an object. And more specifically, if we try using a non-object as the state, it will fails type checking. Specifically because of this:
<img width="659" alt="Screen Shot 2022-10-19 at 1 10 35 PM" src="https://user-images.githubusercontent.com/10232835/196794023-dc7bd9b2-4b47-48cd-9e05-d11b98f63b4d.png">

Reply:
But I can change the state attribute name, to avoid `this.state.state`